opts/network_test.go

```diff
@@ -0,0 +1,122 @@
+package opts
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/pkg/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNetworkOptLegacySyntax(t *testing.T) {
+	testCases := []struct {
+		value    string
+		expected []NetworkAttachmentOpts
+	}{
+		{
+			value: "docknet1",
+			expected: []NetworkAttachmentOpts{
+				{
+					Target: "docknet1",
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		var network NetworkOpt
+		assert.NoError(t, network.Set(tc.value))
+		assert.Len(t, network.Value(), len(tc.expected))
+		for _, expectedNetConfig := range tc.expected {
+			verifyNetworkOpt(t, network.Value(), expectedNetConfig)
+		}
+	}
+}
+
+func TestNetworkOptCompleteSyntax(t *testing.T) {
+	testCases := []struct {
+		value    string
+		expected []NetworkAttachmentOpts
+	}{
+		{
+			value: "name=docknet1,alias=web,driver-opt=field1=value1",
+			expected: []NetworkAttachmentOpts{
+				{
+					Target:  "docknet1",
+					Aliases: []string{"web"},
+					DriverOpts: map[string]string{
+						"field1": "value1",
+					},
+				},
+			},
+		},
+		{
+			value: "name=docknet1,alias=web1,alias=web2,driver-opt=field1=value1,driver-opt=field2=value2",
+			expected: []NetworkAttachmentOpts{
+				{
+					Target:  "docknet1",
+					Aliases: []string{"web1", "web2"},
+					DriverOpts: map[string]string{
+						"field1": "value1",
+						"field2": "value2",
+					},
+				},
+			},
+		},
+		{
+			value: "name=docknet1",
+			expected: []NetworkAttachmentOpts{
+				{
+					Target:  "docknet1",
+					Aliases: []string{},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		var network NetworkOpt
+		assert.NoError(t, network.Set(tc.value))
+		assert.Len(t, network.Value(), len(tc.expected))
+		for _, expectedNetConfig := range tc.expected {
+			verifyNetworkOpt(t, network.Value(), expectedNetConfig)
+		}
```

Comment:
Won't the parsed values be in the same order as `expected`? If so, this loop can be simplified to:

```go
assert.Equal(t, expectedNetConfig, network.Value())
```

and `verifyNetworkOpt` can be removed.